core/src/main/scala/filodb.core/RateLimiter.scala

```diff
@@ -20,7 +20,8 @@ class RateLimiter(period: Duration) {
    *   may be acceptable in some use-cases).
    */
   def attempt(): Boolean = {
-    val nowMillis = System.currentTimeMillis()
+    // Wrapping in a max() because System.currentTimeMillis() is not monotonic.
+    val nowMillis = Math.max(System.currentTimeMillis(), lastSuccessMillis)
```

Comment:
Use ```System.nanoTime```

Reply:
I generally avoid that because I know there is a performance cost, but I've made the change and removed the `max` (assumption is that the cost difference will be negligible in practice).